openmdao/docs/advanced_guide/implicit_comps/defining_icomps.rst

```diff
@@ -0,0 +1,160 @@
+*********************************************************************
+Building Models with Implicit Components
+*********************************************************************
+
+This tutorial will show you how to define implicit components and build models with them.
+We'll work through them using a nonlinear circuit analysis example problem.
+
+Circuit analysis
+********************
+
+Consider a simple electrical circuit made up from two resistors, a diode, and a constant current source.
+Our goal is to solve for the steady-state voltages at node 1 and node 2.
+
+.. figure:: images/circuit_diagram.png
+   :align: center
+   :width: 50%
+   :alt: diagram of a simple circuit with two resistors and one diode
+
+In order to find the voltages we'll employ Kirchoff's law and solve for the voltages needed at each node to drive the net-current to 0.
+
+This means that the voltages at each node are *state variables* for the analysis.
+In other words, V1 and V2 are defined implicitly by the following residual equation:
+
+.. math::
+
+   \mathcal{R_{node_j}} = \sum_k I_{k}^{in} - \sum_k I_{k}^{out} = 0 .
+
+To build this model we're going to define three different components:
+
+    #. Resistor (Explicit)
+    #. Diode (Explicit)
+    #. Node (Implicit)
+
+ExplicitComponents - Resistor and Diode
+***************************************
+
+The :code:`Resistor` and :code:`Diode` components will each compute their current, given the voltages on side.
+These calculations are analytic functions, so we'll inherit from :ref:`ExplicitComponent <comp-type-2-explicitcomp>`.
+These components will each declare some metadata to allow you to pass in the relevant physical constants and give some reasonable default values.
+
+.. embed-code::
+     openmdao.test_suite.test_examples.test_circuit_analysis.Resistor
+
+.. embed-code::
+     openmdao.test_suite.test_examples.test_circuit_analysis.Diode
+
+.. note::
+    Since we've provided default values for the metadata, they won't be required arguments when instantiating :code:`Resistor` or :code:`Diode`.
+    Check out the :ref:`Feature Manual <FeatureManual>` for more details on how to use :ref:`component metadata <component_metadata>`.
+
+
+ImplicitComponent - Node
+***************************************
+
+The :code:`Node` component inherits from :ref:`ImplicitComponent <comp-type-3-implicitcomp>`, which has a different interface than :ref:`ExplicitComponent <comp-type-2-explicitcomp>`.
+Rather than compute the values of its outputs, it computes residuals via the :code:`apply_nonlinear` method.
+When those residuals has been driven to zero, the values of the outputs will be implicitly known.
+:code:`apply_nonlinear` populates the :code:`residual` vector using values from  :code:`inputs` and :code:`outputs` vectors.
```

Comment:
I still think the use of the term `vector` is inappropriate here...    I think you could just say: `The apply_nonlinear function calculates the residuals based on the input and output values`.